src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/ConditionalWeakTable.cs

```diff
@@ -550,8 +552,8 @@ internal bool TryGetEntry(int index, out TKey key, out TValue value)
                     }
                 }
 
-                key = default;
-                value = default;
+                key = default!; // TODO-NULLABLE-GENERIC
```

Comment:
This seem like a bug right? because TKey and TValue are constrained to `class?` ?

Reply:
> This seem like a bug right? because TKey and TValue are constrained to class? ?

Unfortunately, no, this is part of the whole [MaybeNull] discussion.  The `class?` constraint just means that the caller can use a nullable type as the T, but it doesn't mean they must.

Reply:
😢